### PR TITLE
Improve tabview docs

### PR DESCRIPTION
**Pull request by Moosems**

### CODE REVIEW

**Comment by Moosems** [comment]
@rdbende Could I get a review?

File: src/routes/docs/layouts/tabview.md [hunk 1]
```diff
@@ -29,3 +29,5 @@ button_2.grid(margin=20)
 app.run()
 
 ```
+
+The first two blocks are fairly typical for a `Tukaan` program with the necessary import statement and app/window creation. The next block is where the `TabView` is clearly made and is also fairly standard. The next block, however, is slightly unorthodox. In this block we do the following: `tab_x = tabview.Tab("Tab x")`. Digging into the `TabView` source code we find [`self.Tab = Tab`](https://github.com/tukaan/tukaan/blob/f39bb197b1bbbb05a03e498114cc1c4dcff8b800/tukaan/widgets/tabview.py#L152-L155). If we go up in the file we then find class `Tab` which is really just a wrapper around the `Frame` class with extra methods for tabs. Because of how `Tab` is implemented, however, it creates a very clean API call that doesn't require much extra thought to understand. Moving on to the next two blocks we see that these tabs are interacted with exactly as one would work with a `Frame` which of course makes sense knowing that they are indeed a Frame on the inside. The final block is also fairly generic with the call of the `App.run()` method.
```
Comment (rdbende) [inline src/routes/docs/layouts/tabview.md]: - I don't think it's necessary t explain every part of the code. At this point the reader is expected to know the parts where you use "fairly standard/generic", and I don't want the docs to re-explain these basic patterns of Tukaan everywhere.

- Please don't explain what the source code does, it's irrelevant. All the user should care about here is how they does things with the public API, the rest is implementation detail.

I think what this part should be about is the you can create a new tab by creating a `.Tab()`, and it works like an other container. The user can set parameters of the tab, like the title and icon, and can call different methods on it, like `.move()`, `.hide()`, or `.remove()`

File: src/routes/docs/layouts/tabview.md [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 ![Image of a Tukaan tabview](/resources/docs/tabview.png)
 
-An example of how to set up a TabView widget is as follows:
+The `TabView` widget allows users to have multiple tabs (which are ontainers) in which they can place widgets. An example of how to set up a TabView widget is as follows:
```
Comment (rdbende) [inline src/routes/docs/layouts/tabview.md]: I'd like to use "you" to address the reader.
There's a typo, too.

**Review by rdbende — CHANGES_REQUESTED** [review]

File: src/routes/docs/layouts/tabview.md (hunk 1 above)
Comment (Moosems) [reply src/routes/docs/layouts/tabview.md]: Got it 👍

**Review by Moosems — COMMENTED** [review]
Update docs

**Comment by Moosems** [comment]
Unfortunate that the page took an hour to rewrite but I'm happy with how it turned out

File: src/routes/docs/layouts/tabview.md [hunk 3]
```diff
@@ -27,5 +27,42 @@ button_2 = tukaan.Button(tab_2, "Button in tab 2")
...
+| `margin` | `int` or `tuple[int, ...]` | Sets the margins of the `Tab` |
+
+## `TabView` Options
+
+The `TabView` also has a couple useful methods. They go as follows:
+
+| Method| Description|
+|-|-|
+| `.on_tab_change(func: Callable[[Tab | None], None]) -> Callable[[], None]` | Binds a function that is called any time tab is changed |
+| `.enable_keyboard_traversal()` | Enables traversal o
```
Comment (rdbende) [inline src/routes/docs/layouts/tabview.md]:
```suggestion
| `.enable_keyboard_traversal()` | Enables navigating the tabs with the keyboard
```

File: src/routes/docs/layouts/tabview.md [hunk 4]
```diff
@@ -27,5 +27,42 @@ button_2 = tukaan.Button(tab_2, "Button in tab 2")
 button_2.grid(margin=20)
 
 app.run()
-
 ```
+
+## `Tab` Options
+
+The `TabView` widget allows users to create `Tab`'s as shown above that act as `Container`'s. The unique methods available to `Tab`'s and a description of each method are as follows:
```
Comment (rdbende) [inline src/routes/docs/layouts/tabview.md]:
```suggestion
The `TabView` widget allows users to create `Tab`'s as shown above that act as container's. The unique methods available to `Tab`'s and a description of each method are as follows:
```

It's technically a container, but not literally

File: src/routes/docs/layouts/tabview.md [hunk 5]
```diff
@@ -2,7 +2,7 @@
 
 ![Image of a Tukaan tabview](/resources/docs/tabview.png)
 
-An example of how to set up a TabView widget is as follows:
+The `TabView` widget allows you to have multiple tabs where you can place widgets like any other `Container`. An example of how to set up a TabView widget is as follows:
```
Comment (rdbende) [inline src/routes/docs/layouts/tabview.md]:
```suggestion
The `TabView` widget allows you to have multiple tabs where you can place widgets like any other container widget. An example of how to set up a TabView widget is as follows:
```

File: src/routes/docs/layouts/tabview.md [hunk 6]
```diff
@@ -27,5 +27,42 @@ button_2 = tukaan.Button(tab_2, "Button in tab 2")
...
+
+The `TabView` widget allows users to create `Tab`'s as shown above that act as `Container`'s. The unique methods available to `Tab`'s and a description of each method are as follows:
+
+| Method | Description|
+|-|-|
+| `.select()` | Sets the `Tab` showing in the `TabView` to the `Tab` this is called on |
+| `.move(new_index: int)` | Moves the `Tab` to the specified index in the ordering (left to right) of the `TabView` |
+| `.hide()` | Hides the `Tab` visually but is still considered to be in the `TabView` |
+| `.unhide()` | Shows the `Tab` again where it was before |
+| `.remove()` | Removes the `Tab` from the list internally |
```
Comment (rdbende) [inline src/routes/docs/layouts/tabview.md]:
```suggestion
| `.remove()` | Removes the `Tab` from the `TabView`, and can only be shown again by calling `.append()` |
```

**Comment by rdbende** [comment]
> Unfortunate that the page took an hour to rewrite but I'm happy with how it turned out

I think it's pretty good as well!

**Review by rdbende — APPROVED** [review]

**Pull request merged**